### PR TITLE
Add some tests for the user benefits handler

### PR DESCRIPTION
**Pull request by tjmw**
## What does this change?

I'm planning on making some changes to the user benefits handler, including to the routing, so add some tests. This involved a little bit of setup to make jest work in this workspace.

## How to test

```
$ pnpm --filter user-benefits test
```

## Accessibility

-   [ ] [Tested with screen reader](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#screen-reader)
-   [ ] [Navigable with keyboard](https://github.com/guardian/accessibility/blob/main/people-and-technology/02-physical.md#keyboard)
-   [ ] [Colour contrast passed](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#contrast)
-   [ ] [The change doesn't use only colour to convey meaning](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#use-of-colour)

### CODE REVIEW

File: modules/identity/src/apiGateway.ts [hunk 1]
```diff
@@ -84,3 +84,17 @@ export class IdentityApiGatewayAuthenticator {
 		}
 	}
 }
+
+export const buildAuthenticate = (
```
Comment (tjmw) [inline modules/identity/src/apiGateway.ts:88]: I've added this to simplify the interface from the user benefits handler perspected (it makes mocking a bit easier).

File: handlers/user-benefits/src/index.ts [hunk 2]
```diff
@@ -44,7 +40,7 @@ const getUserBenefitsResponse = async (
 	};
 };
 
-export const handler: Handler = async (
```
Comment (tjmw) [inline handlers/user-benefits/src/index.ts:47]: I've removed this so we can be slightly looser in the tests about the args specified (strictly speaking according to the `Handler` definition, we should invoke `handler` with two additional args which we don't actually need). This is consistent with other lambda definitions in this repo.

**Review by rupertbates — APPROVED** [review]

File: handlers/user-benefits/test/index.test.ts [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+}));
+jest.mock('@modules/product-catalog/api', () => ({
+	getProductCatalogFromApi: () => ({}),
+}));
+jest.mock('@modules/product-benefits/userBenefits', () => ({
+	getUserBenefits: () => ['adFree'],
+}));
+
+describe('handler', () => {
+	it('returns a 404 for an unrecognized path or HTTP method', async () => {
```
Comment (rupertbates) [inline handlers/user-benefits/test/index.test.ts:31]: Probably don't need this now, it's already tested in the router module

Comment (tjmw) [reply handlers/user-benefits/test/index.test.ts:31]: Discussed offline: we're not migrated over to the router module yet, but that's the next step (and I'll remove this test then).

**Pull request merged**